README.md

```diff
@@ -19,23 +21,149 @@ let data = T::deserialize_from_value(json.into_value()).unwrap();
+can do is generate a code `bad_request` that is common for our whole API. We then use this code to generate
+a link to our documentation to help our users. But how do you want to help anyone with an error
+that absolutely every single route of meilisearch can throw?
+
+#### With deserr
+
+```json
+{
+  "message": "Invalid value at `.typoTolerance.minWordSizeForTypos.oneTypo`: value: `1000` is too large to be deserialized, maximum value authorized is `255`",
+  "code": "invalid_settings_typo_tolerance",
+  "type": "invalid_request",
+  "link": "https://docs.meilisearch.com/errors#invalid-settings-typo-tolerance"
+}
+```
+
+##### The message
+
+> Invalid value at `.typoTolerance.minWordSizeForTypos.oneTypo`: value: `1000` is too large to be deserialized, maximum value authorized is `255`
+
+- We get a more human-readable location; `.typoTolerance.minWordSizeForTypos.oneTypo`. It gives us the faulty field.
+- We also get a non-confusing and helpful message this time; it explicitly tells us that the maximum value authorized is `255`.
+
+##### The rest of the payload
+
+Since deserr called one of our functions in the process, we were able to use a custom error code + link to redirect
+our user to the documentation specific to this feature and this field.
+
+#### More possibilities with deserr that were impossible with serde
+
+##### Adding constraints on multiples fields
+
+In meilisearch, there is another constraint on this `minWordSizeForTypos`, the `twoTypo` field **must be** greater than
+the `oneType` field.
+
+Serde doesn't provide any feature to do that. You could write your own implementation of `Deserialize` for the
+entire sub-object `minWordSizeForTypos`, but that's generally hard, and we tend to avoid doing it.
+Thus, that's the kind of thing you're going to check by hand in your code later on. Which may bring
+inconsistencies between most of the deserialization error messages and your error message.
+
+With deserr, we provide attributes that allow you to validate your structure once it's deserialized.
+
+##### When a field is missing
+
+It's possible to provide your own function when a field is missing.
+
+```rust,ignore
+pub fn missing_field<E: DeserializeError>(field: &str, location: ValuePointerRef) -> E {
+    todo!()
+}
+```
+
+At meilisearch, we use this function to specify a custom error code, but we keep the default error message which is pretty accurate.
+
+##### When an unknown field is encountered
+
+It's possible to provide your own function when a field is missing.
+
+```rust,ignore
+fn unknown_field<E: DeserializeError>(
+    field: &str,
+    accepted: &[&str],
+    location: ValuePointerRef,
+) -> E {
+    todo!()
+}
+```
+
+Here is a few ideas we have or would like to implement at meilisearch;
+- Detecting when you try to `PATCH` a field that doesn't exists to throw a special `immutable field x` error instead of an `unknown field x`
```

Comment:
Oh yeah, I'm thinking about a field that is present during a `POST`, but not present during a `PATCH` because you can't update it (like we have for our API keys).
But yeah, it's not super clear :thinking: